core/trino-spi/src/main/java/io/trino/spi/block/BlockUtil.java

```diff
@@ -371,24 +363,11 @@ static void appendRawBlockRange(Block rawBlock, int offset, int length, BlockBui
         }
     }
 
-    /**
-     * Ideally, the underlying nulls array in Block implementations should be a byte array instead of a boolean array.
-     * This method is used to perform that conversion until the Block implementations are changed.
-     */
-    static Optional<ByteArrayBlock> getNulls(@Nullable boolean[] valueIsNull, int arrayOffset, int positionCount)
+    static Optional<ByteArrayBlock> getNulls(@Nullable byte[] valueIsNull, int arrayOffset, int positionCount)
```

Comment:
Nice, please also simplify IsNullColumnarFilter and IsNotNullColumnarFilter to not do the extra mayHaveNull check. I realised now that it was unnecessary.